### PR TITLE
rocketmq 4.9.0 (new formula)

### LINKED ISSUE #19295: Suggestion:add the rocketmq
when the rocketmq became apache top level project,i want to install it by brew install rocketmq.but i can't find it.
-----------------------------------------------------
sorry,I just started using honwbrew,doesn't know the Formulae syntax

### PR DESCRIPTION
**Pull request by messense**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

Fixes https://github.com/Homebrew/homebrew-core/issues/19295

### CODE REVIEW

File: Formula/rocketmq.rb [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+      s.gsub! "JAVA_OPT=\"${JAVA_OPT} -XX:-UseLargePages -XX:-UseBiasedLocking\"",
+        "JAVA_OPT=\"${JAVA_OPT} -XX:-UseLargePages\""
+      s.gsub! "choose_gc_log_directory\n", "\n"
+    end
+
+    inreplace "bin/tools.sh" do |s|
+      s.gsub! "export CLASSPATH=.:${BASE_DIR}/conf:${CLASSPATH}",
+        "export CLASSPATH=.:${BASE_DIR}/lib/*:${BASE_DIR}/conf:${CLASSPATH}"
+      s.gsub! "JAVA_OPT=\"${JAVA_OPT} -Djava.ext.dirs", "#JAVA_OPT=\"${JAVA_OPT} -Djava.ext.dirs"
+    end
```
Comment (SMillerDev) [inline Formula/rocketmq.rb:41]: New formulae should not require patches to work. Please submit these fixes upstream.

https://docs.brew.sh/Acceptable-Formulae#niche-or-self-submitted-stuff

File: Formula/rocketmq.rb [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+
+    # netty-all-4.0.42.Final.jar doesn't work on arm64 macOS
+    rm_rf "lib/netty-all-4.0.42.Final.jar"
+    libexec.install "lib"
+
+    # install netty-all-4.1.65.Final.jar which works on arm64 macOS
+    resource("netty-all").stage { cp "netty-all-4.1.65.Final.jar", libexec/"lib" }
```
Comment (SMillerDev) [inline Formula/rocketmq.rb:48]: This should be fixed upstream

File: Formula/rocketmq.rb [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+      brokerId = 0
+      deleteWhen = 04
+      fileReservedTime = 48
+      brokerRole = ASYNC_MASTER
+      flushDiskType = ASYNC_FLUSH
+      storePathRootDir=#{var}/rocketmq
+      storePathCommitLog=#{var}/rocketmq/commitlog
+      autoCreateTopicEnable=true
+    EOS
+  end
```
Comment (SMillerDev) [inline Formula/rocketmq.rb:93]: We don't want to ship configuration if upstream doesn't

File: Formula/rocketmq.rb [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+  def caveats
+    <<~EOS
+      Data:    #{var}/rocketmq/commitlog
+      Logs:    #{var}/log/rocketmq
+      Config:  #{etc}/rocketmq/broker.conf
+
+      Start Apache RocketMQ server with two steps:
+      1. mqnamesrv
+      2. mqbroker -n localhost:9876
+    EOS
```
Comment (SMillerDev) [inline Formula/rocketmq.rb:104]: This should be in the upstream docs, not a brew caveat.

File: Formula/rocketmq.rb [hunk 5]
```diff
@@ -0,0 +1,133 @@
+class Rocketmq < Formula
+  desc "Distributed messaging and streaming platform"
+  homepage "https://rocketmq.apache.org"
+  url "https://www.apache.org/dyn/closer.lua?path=rocketmq/4.9.0/rocketmq-all-4.9.0-bin-release.zip"
+  mirror "https://archive.apache.org/dist/rocketmq/4.9.0/rocketmq-all-4.9.0-bin-release.zip"
```
Comment (SMillerDev) [inline Formula/rocketmq.rb:5]: Does this include any precompiled binaries?

File: Formula/rocketmq.rb (hunk 2 above)
Comment (messense) [reply Formula/rocketmq.rb:48]: Submitted https://github.com/apache/rocketmq/pull/3032

File: Formula/rocketmq.rb (hunk 5 above)
Comment (messense) [reply Formula/rocketmq.rb:5]: It only includes precompiled Java cross-platform binaries.

File: Formula/rocketmq.rb (hunk 3 above)
Comment (messense) [reply Formula/rocketmq.rb:93]: Upstream does ship configuration, added `storePathRootDir`, `storePathCommitLog` path to fit Homebrew.

**Review by messense — COMMENTED** [review]
It will take some time for upstream to accept patches, it's fine to close this PR in the meantime.

**Pull request closed without merging**